crossfire/__init__.py

```diff
@@ -70,6 +70,9 @@ def flatten(data, nested_columns=None):
     nested_columns = set(nested_columns or NESTED_COLUMNS)
     if not nested_columns.issubset(NESTED_COLUMNS):
         raise NestedColumnError(nested_columns)
+    if not HAS_PANDAS or (HAS_PANDAS and not isinstance(data, DataFrame)):
+        if not data:
+            return data
```

Comment:
What about this?

Reply:
Because the user can have pandas but for a specific reason the `data` is not a `DataFrame` instance...

Reply:
This is better. But a lot is going on, and it is quite a repetitive sequence of checks between lines 73-78 (and you still have two early returns for the same case, empty `data`, one on line 75, another on line 78). Have you considered a function like `is_empty(data)` that returns a boolean to that question?

Reply:
> This is better. But a lot is going on, and it is quite a repetitive sequence of checks between lines 73-78 (and you still have two early returns for the same case, empty `data`, one on line 75, another on line 78). Have you considered a function like `is_empty(data)` that returns a boolean to that question?

Sure I can. I should be done this before. By the way... Should I be writting tests for `flatten_df`, and `is_empty`, right?